dytype/test_dytype.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+assert_unspecified() {
+  if grep --quiet ': #<unspecified>' <(./dytype "$@"); then
```

Comment:
This and `assert_n` should probably use regex `$` to ensure it ends with the correct type.